test/integration/controller/job/job_controller_test.go

```diff
@@ -265,6 +268,217 @@ var _ = ginkgo.Describe("Job controller for workloads with no queue set", func()
 	})
 })
 
+var _ = ginkgo.Describe("Job controller when waitForPodsReady enabled", func() {
```

Comment:
Can we add a test to verify that the condition gets updated for a workload that initially has `spec.Admission` set and then unset (to simulate preemption).

Reply:
Modified the tests. Now I unconditionally admit the workload, but if a test specifies "suspended=true" then Admission is unset and the job is suspended.

Reply:
sg, we can have a test in the scheduler once preemption is implemented.